yt/loaders.py

```diff
@@ -1263,7 +1264,9 @@ def flatten(l):
 
 
 # --- Loader for yt sample datasets ---
-def load_sample(fn=None, progressbar: bool = True, timeout=None, **kwargs):
+def load_sample(
+    fn: Optional[str] = None, progressbar: bool = True, timeout=None, **kwargs
```

Comment:
Genuine question: why are we marking this optional if it's also a kwarg, and if the others are not explicitly marked optional?

Reply:
Alas, `typing.Optional[str]` strictly means `str | None` (where "|" reads as "or"). The later syntax is only valid from Python 3.10 (not released) though. Arguably this is a design mistake from early versions of the `typing` module.